lib/node_modules/@stdlib/array/base/map/test/test.assign.js

```diff
@@ -0,0 +1,213 @@
+var map = require( './../lib' ).assign;
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof map, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function applies a provided callback to elements in an input array and assigns results to elements in an output array (generic)', function test( t ) {
+	var expected;
+	var out;
+	var x;
+	var y;
+
+	x = [ 1, 2, 3, 4 ];
+	y = zeros( x.length );
+	expected = [ 10, 20, 30, 40 ];
+
+	out = map( x, y, 1, 0, scale );
+
+	t.deepEqual( out, expected, 'returns expected value' );
+	t.strictEqual( out, y, 'returns expected value' );
+	t.end();
+
+	function scale( v ) {
+		return v * 10;
+	}
+});
+
+tape( 'the function applies a provided callback to elements in an input array and assigns results to elements in an output array (typed array)', function test( t ) {
+	var expected;
+	var out;
+	var x;
+	var y;
+
+	x = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+	y = new Float64Array( x.length );
+	expected = new Float64Array( [ 10.0, 20.0, 30.0, 40.0 ] );
+
+	out = map( x, y, 1, 0, scale );
+
+	t.deepEqual( out, expected, 'returns expected value' );
+	t.strictEqual( out, y, 'returns expected value' );
+	t.end();
+
+	function scale( v ) {
+		return v * 10.0;
+	}
+});
+
+tape( 'the function applies a provided callback to elements in an input array and assigns results to elements in an output array (accessors)', function test( t ) {
+	var expected;
+	var out;
+	var x;
+	var y;
+
+	x = [ 1, 2, 3, 4 ];
+	y = zeros( x.length );
+	expected = toAccessorArray( [ 10, 20, 30, 40 ] );
+
+	out = map( toAccessorArray( x ), toAccessorArray( y ), 1, 0, scale );
+
+	t.deepEqual( out, expected, 'returns expected value' );
+	t.end();
+
+	function scale( v ) {
+		return v * 10;
+	}
+});
+
+tape( 'the function applies a provided callback to elements in an input array and assigns results to elements in an output array (array-like object)', function test( t ) {
+	var expected;
+	var out;
+	var x;
+	var y;
+
+	x = {
+		'length': 4,
+		'0': 1,
+		'1': 2,
+		'2': 3,
+		'3': 4
+	};
+	y = zeros( x.length );
+	expected = [ 10, 20, 30, 40 ];
+
+	out = map( x, y, 1, 0, scale );
+
+	t.deepEqual( out, expected, 'returns expected value' );
+	t.strictEqual( out, y, 'returns expected value' );
+	t.end();
+
+	function scale( v ) {
+		return v * 10;
+	}
+});
+
+tape( 'the function supports providing an execution context', function test( t ) {
+	var expected;
+	var out;
+	var ctx;
+	var x;
+	var y;
+
+	x = [ 1, 2, 3, 4 ];
+	y = zeros( x.length );
+	ctx = {
+		'factor': 10
+	};
+	expected = [ 10, 20, 30, 40 ];
+
+	out = map( x, y, 1, 0, scale, ctx );
+
+	t.deepEqual( out, expected, 'returns expected value' );
+	t.strictEqual( out, y, 'returns expected value' );
+	t.end();
+
+	function scale( v ) {
+		return v * this.factor; // eslint-disable-line no-invalid-this
+	}
+});
+
+tape( 'the function supports a `stride` parameter', function test( t ) {
```

Comment:
We need to replicate these tests for each input argument type: "generic", "accessors", "typed array", and "array-like object". While internally, they may share the same implementation; from a user POV, that is not guaranteed. As such, unit tests should be comprehensive in terms of expected behavior. In general, we should not be writing tests based on insider knowledge.